In [1]:
import pandas as pd

stops = pd.read_csv('data/stops.txt')
schedule = pd.read_csv('data/stop_times.txt')

In [2]:
result = schedule.groupby('stop_id').size().reset_index(name='NoOfSchedules').sort_values(by='NoOfSchedules', ascending=False)
print(result.to_string(index=False))

 stop_id  NoOfSchedules
     600           2787
     420           2685
     575           2544
     558           2368
     538           2001
     450           1995
     430           1899
     468           1862
     515           1847
     480           1819
     400           1723
     605           1716
    1610           1700
   35317           1649
     500           1521
     570           1345
     548           1345
    2220           1342
   11352           1340
   10914           1340
   10912           1340
     590           1335
   10911           1226
     490           1221
    1085           1208
   30635           1208
    3600           1129
   10917           1115
    8402           1111
    1480           1104
   29240           1095
    1530           1066
    1471           1056
     531           1054
     431           1048
   81301           1037
    9587           1028
   41740           1004
    1490           1004
    1500            994
    2330        

In [3]:
schedule[schedule["stop_id"] == 31135]

trip_id arrival_time departure_time  stop_id  stop_sequence  \
106284  571921968     06:59:00       06:59:00    31135            157   

       stop_headsign  pickup_type  drop_off_type  shape_dist_traveled  \
106284           NaN            0              0              27469.8   

        timepoint  
106284          1

trip 571921968 is serviced on weekdays except mon from 1 may to 6 june and few exceptions 
service_id	date	exception_type

`270	30003	20250526	2 //"mon off" no need already off?
271	30003	20250602	1
272	30003	20250519	1
273	30003	20250512	1
274	30003	20250505	1`

In [4]:
result[result["NoOfSchedules"].isin(range(25, 30))]

stop_id  NoOfSchedules
5523    75972             29
5677    78046             29
3698    59033             29
5675    78044             29
4321    64364             29
...       ...            ...
3369    57176             25
3368    57175             25
3367    57174             25
3370    57177             25
3371    57178             25

[170 rows x 2 columns]

In [5]:
schedule[schedule['stop_id'] == 75972].sort_values(by='departure_time')

trip_id arrival_time departure_time  stop_id  stop_sequence  \
12485   472991568     06:18:39       06:18:39    75972            680   
477461  677625238     07:26:39       07:26:39    75972           1383   
732603  710854478     07:53:39       07:53:39    75972           1348   
579201  692131558     07:53:39       07:53:39    75972           1348   
476827  676559028     08:34:39       08:34:39    75972           1383   
732629  710854488     09:36:39       09:36:39    75972           1348   
579227  692131568     09:36:39       09:36:39    75972           1348   
476603  676558948     09:42:39       09:42:39    75972           1383   
477517  677625258     10:50:39       10:50:39    75972           1383   
732655  710854498     11:08:39       11:08:39    75972           1348   
579253  692131578     11:08:39       11:08:39    75972           1348   
476883  676559048     11:58:39       11:58:39    75972           1383   
732681  710854508     12:47:39       12:47:39    75972           1348   
579279  692131588     12:47:39       12:47:39    75972           1348   
476659  676558968     13:06:39       13:06:39    75972           1383   
477573  677625278     14:14:39       14:14:39    75972           1383   
579305  692131598     14:26:39       14:26:39    75972           1348   
732707  710854518     14:26:39       14:26:39    75972           1348   
476939  676565738     15:22:39       15:22:39    75972           1383   
732733  710854528     16:07:39       16:07:39    75972           1348   
579331  692131608     16:07:39       16:07:39    75972           1348   
476715  676558988     16:30:39       16:30:39    75972           1383   
477376  677625198     17:38:39       17:38:39    75972           1383   
579357  692131618     17:52:39       17:52:39    75972           1348   
732759  710854538     17:52:39       17:52:39    75972           1348   
476995  676565758     18:46:39       18:46:39    75972           1383   
579383  692131628     19:38:39       19:38:39    75972           1348   
732785  710854548     19:38:39       19:38:39    75972           1348   
476771  676559008     19:54:39       19:54:39    75972           1383   

       stop_headsign  pickup_type  drop_off_type  shape_dist_traveled  \
12485            NaN            0              0              85846.8   
477461           NaN            0              0             152871.2   
732603           NaN            0              0             148750.9   
579201           NaN            0              0             148750.9   
476827           NaN            0              0             152871.2   
732629           NaN            0              0             148750.9   
579227           NaN            0              0             148750.9   
476603           NaN            0              0             152871.2   
477517           NaN            0              0             152871.2   
732655           NaN            0              0             148750.9   
579253           NaN            0              0             148750.9   
476883           NaN            0              0             152871.2   
732681           NaN            0              0             148750.9   
579279           NaN            0              0             148750.9   
476659           NaN            0              0             152871.2   
477573           NaN            0              0             152871.2   
579305           NaN            0              0             148750.9   
732707           NaN            0              0             148750.9   
476939           NaN            0              0             152871.2   
732733           NaN            0              0             148750.9   
579331           NaN            0              0             148750.9   
476715           NaN            0              0             152871.2   
477376           NaN            0              0             152871.2   
579357           NaN            0              0             148750

In [6]:
ss = schedule[schedule["stop_id"]==600]
ss[ss["departure_time"] >= "24:00:00"].sort_values(by='departure_time')

trip_id arrival_time departure_time  stop_id  stop_sequence  \
620789  694026798     24:00:09       24:00:09      600             68   
780436  721325928     24:00:30       24:00:30      600             68   
776721  721293498     24:00:30       24:00:30      600             68   
604737  693991538     24:01:32       24:01:32      600             68   
454673  675344448     24:02:00       24:02:00      600            123   
...           ...          ...            ...      ...            ...   
571169  692096268     27:37:52       27:37:52      600            417   
299902  636332878     27:38:00       27:38:00      600            462   
898291  727423328     27:42:00       27:42:00      600            462   
462157  675347168     28:36:00       28:36:00      600             97   
898477  727423408     28:42:00       28:42:00      600            462   

           stop_headsign  pickup_type  drop_off_type  shape_dist_traveled  \
620789               NaN            0              0               8221.8   
780436               NaN            0              0               8221.8   
776721               NaN            0              0               8221.8   
604737               NaN            0              0               8221.8   
454673               NaN            0              0              17801.0   
...                  ...          ...            ...                  ...   
571169               NaN            0              0              59167.4   
299902  Downtown Seattle            0              0              68941.0   
898291  Downtown Seattle            0              0              68941.0   
462157               NaN            0              0              13625.0   
898477  Downtown Seattle            0              0              68941.0   

        timepoint  
620789          0  
780436          0  
776721          0  
604737          0  
454673          1  
...           ...  
571169          0  
299902          1  
898291          1  
462157          1  
898477          1  

[146 rows x 10 columns]